src/map/utils/battleutils.cpp

```diff
@@ -2140,7 +2140,7 @@ namespace battleutils
 
                         // Shield Mastery
                         if ((std::max(damage - (PDefender->getMod(Mod::PHALANX) + PDefender->getMod(Mod::STONESKIN)), 0) > 0) &&
-                            charutils::hasTrait((CCharEntity*)PDefender, TRAIT_SHIELD_MASTERY))
+                            PDefender->getMod(Mod::SHIELD_MASTERY_TP))
```

Comment:
The other similar checks are looking at the mod not the trait which makes sense since you don't necessarily need to have the trait for the equipment bonus to apply. For example: https://www.bg-wiki.com/ffxi/Adoubeur%27s_Pavise

Reply:
Would make sense as this is similar to `STORE_TP`, which doesn't require the job trait to provide the bonus

Reply:
Got confirmation:
![image](https://github.com/LandSandBoat/server/assets/131182600/337ff20e-c669-43c5-a2f9-adc744df4ad6)